libs/native/src/main/java/org/elasticsearch/nativeaccess/NoopNativeAccess.java

```diff
@@ -8,15 +8,24 @@
 
 package org.elasticsearch.nativeaccess;
 
-class NoopNativeAccess extends AbstractNativeAccess {
+import org.elasticsearch.logging.LogManager;
+import org.elasticsearch.logging.Logger;
 
-    NoopNativeAccess() {
-        super("noop");
-    }
+class NoopNativeAccess implements NativeAccess {
+
+    private static final Logger logger = LogManager.getLogger(NativeAccess.class);
+
+    NoopNativeAccess() {}
 
     @Override
     public boolean definitelyRunningAsRoot() {
         logger.warn("Cannot check if running as root because native access is not available");
         return false;
     }
+
+    @Override
+    public Zstd getZstd() {
+        logger.warn("cannot compress with zstd because native access is not available");
+        return null;
```

Comment:
When ZSTD support becomes a requirement in the future, I imagine we'd throw here instead?

Reply:
Yes, ideally there would be no no-op implementation, we would throw at load time.

Reply:
Is there a reason we aren't just doing that now?

Reply:
It's contentious, so I want to save that for an isolated discussion/PR.

Reply:
This sounds like leniency, which I presumed we had all agreed is abhorrent 😉